pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -250,7 +250,24 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
     }
     String query = sql.asText();
     requestContext.setQuery(query);
-    return handleRequest(requestId, query, sqlNodeAndOptions, request, requesterIdentity, requestContext);
+    BrokerResponse brokerResponse = handleRequest(requestId, query, sqlNodeAndOptions, request,
+            requesterIdentity, requestContext);
+
+    boolean dropResults = false;
+    if (request.has(Broker.Request.QUERY_OPTIONS)) {
+      String queryOptions = request.get(Broker.Request.QUERY_OPTIONS).asText();
+      Map<String, String> optionsFromString = RequestUtils.getOptionsFromString(queryOptions);
+      if (optionsFromString.getOrDefault(Broker.Request.QueryOptionKey.DROP_RESULTS, "false")
```

Comment:
You may put this logic into `QueryOptionsUtils`

Reply:
It is equivalent to `Boolean.parseBoolean(queryOptions.get(QueryOptionKey.DROP_RESULTS))`